File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        // HashMap<Character, Integer> map = new HashMap<>();

        // for(char c: magazine.toCharArray())
        // {
        //     map.put(c, map.getOrDefault(c, 0) + 1);
        // }

        // for(char c: ransomNote.toCharArray())
        // {
        //     if(map.containsKey(c) && map.get(c) > 0)
        //     {
        //         map.put(c, map.get(c) - 1);
        //     }
        //     else
        //     {
        //         return false;
        //     }
        // }

        // return true;

        // ---------------------------------------------------------

        short[] arr = new short[26];

        for(char c: magazine.toCharArray())
        {
            arr[c - 'a']++;
        }

        for(char c: ransomNote.toCharArray())
        {
            if(arr[c - 'a'] <= 0)
            {
                return false;
            }

            arr[c - 'a']--;
        }

        return true;
    }
}

File: 0383-ransom-note/0383-ransom-note.c
bool canConstruct(char* ransomNote, char* magazine) {
    int count[26] = {0};

    // 统计 magazine 中每个字母的数量
    for(int i = 0 ; magazine[i] != '\0' ; i++){
        count[magazine[i] - 'a']++;
    }

    // 检查 randomNote 中每个字母是否足够
    for(int i = 0 ; ransomNote[i] != '\0' ; i++){
        if(--count[ransomNote[i] - 'a'] < 0){
            return false;
        }       
    }

    return true;
}